src/app/search/state/search.service.spec.ts

```diff
@@ -103,4 +104,46 @@ describe('SearchService', () => {
       expect(workflowSource.imgProviderUrl).toBe(undefined);
     }
   ));
+  it('should sort workflows correctly', inject([SearchService], (service: SearchService) => {
+    const a: Workflow = {
+      author: 'a',
+      gitUrl: 'https://giturl',
+      mode: 'FULL',
+      organization: '',
+      repository: '',
+      sourceControl: '',
+      descriptorType: 'CWL',
+      workflow_path: '',
+      defaultTestParameterFilePath: ''
+    };
+
+    const b: Workflow = {
+      author: 'B',
+      gitUrl: 'https://giturl',
+      mode: 'FULL',
+      organization: '',
+      repository: '',
+      sourceControl: '',
+      descriptorType: 'CWL',
+      workflow_path: '',
+      defaultTestParameterFilePath: ''
+    };
+
+    const c: Workflow = {
+      author: null,
+      gitUrl: 'https://giturl',
+      mode: 'FULL',
+      organization: '',
+      repository: '',
+      sourceControl: '',
+      descriptorType: 'CWL',
+      workflow_path: '',
+      defaultTestParameterFilePath: ''
+    };
+
+    expect(SearchService.compareAttributes(a, b, 'author', 'asc')).toEqual(-1);
```

Comment:
Since  compareAttributes has branching logic based on the attribute name, your tests should cover an attribute that doesn't go into that code branch.

Reply:
added 2 cases for descriptorType attribute